src/bp/lang-server/service/index.ts

```diff
@@ -19,6 +25,14 @@ const FAST_TEXT_MODEL_REGEX = /^(\w+)\.(\w+)\.(\d+)\.bin$/i
 // Examples: "scope.en.bpe.model" "bp.en.bpe.model"
 const BPE_MODEL_REGEX = /^(\w+)\.(\w+)\.bpe\.model$/i
 
+const checkEnoughRam = (langNb: number, dims: number): RamInfos => {
+  return {
+    used: Number.parseFloat((os.freemem() / (1024 * 1024 * 1024)).toFixed(2)),
+    total: Number.parseFloat((os.totalmem() / (1024 * 1024 * 1024)).toFixed(2)),
+    prediction: (dims > 101 ? 4.0 : 2.0) * langNb
```

Comment:
This line is the core logic of this PR.

Idea: Document prediction thresholds/factors more verbosely in code and possibly also in the warning itself. Afaik requirements vary from 1 to 6GB per language depending on the language and dimensions used (25, 100, 300).
E.g. create constants for the values "4.0" and "2.0" to be more clearly readable/adaptable to be tweaked - maybe 5.0 and 1.5 fit better?

Reply:
A next step could be to have a map per language & dims -> size. I will add something like that to be more precise because all models are not equals. 

The problem is that `os.freemem()` is giving the free memory (including the virtual one) so much less than the "real" available one.  Thus it will trigger the warning even if there is sometimes enough memory.

I'm still looking for a better solution to get the "real" ram available but node don't seem to do such Os things and we don't want to add a 50Mb library just for it.